### PR TITLE
feat: add multiple VMs for migration

### PR DESCRIPTION
**Pull request by shakyav**
## Summary by CodeRabbit

- Updates OpenShift CI migration tests to create and migrate multiple VMs in one job.
- Configures the ACM virtualization interoperability job to use five VMs, 6-hour migration timeouts, and optional SSH verification.
- Preserves single-VM naming and adds indexed names for multiple VMs.
- Adds per-VM readiness checks, cleanup, diagnostics, status collection, destination validation, and migration monitoring.
- Extends VM creation timeout to 4 hours.
- Enforces strict `subctl` version validation and digest checks for immutable releases while allowing rolling `release-X.Y` tags without digest verification.

Passing job - https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/83908/rehearse-83908-periodic-ci-RedHatQE-interop-testing-master-acm-virt-ocp4.22-p2p-cclm-liv-mig-lp-interop-aws/2091954692849405952

### CODE REVIEW

**Comment by shakyav** [comment]
/pj-rehearse periodic-ci-RedHatQE-interop-testing-master-acm-virt-ocp4.22-p2p-cclm-liv-mig-lp-interop-aws

**Comment by shakyav** [comment]
/pj-rehearse periodic-ci-RedHatQE-interop-testing-master-acm-virt-ocp4.22-p2p-cclm-liv-mig-lp-interop-aws

**Comment by shakyav** [comment]
/pj-rehearse periodic-ci-RedHatQE-interop-testing-master-acm-virt-ocp4.22-p2p-cclm-liv-mig-lp-interop-aws

File: ci-operator/step-registry/p2p/mtv-execute-live-migration/p2p-mtv-execute-live-migration-commands.sh [hunk 1]
```diff
@@ -561,20 +632,89 @@ WaitMigrationSucceeded() {
...
+          [[ -n "${vmIp}" ]] || exit 2
+          DestOc exec -n "${targetNs}" "${anchorPod}" -c compute -- \
+              timeout 15 bash -c "echo > /dev/tcp/${vmIp}/22"
+        ) || probeRc=$?
+
+        case "${probeRc}" in
+            0) : "SSH port 22 reachable on ${vmName}" ;;
+            2) : "No IP on VMI ${vmName}; skipping SSH probe" ;;
+            *) : "SSH port 22 not reachable on ${vmName} (rc=${probeRc})"
+               (( failed++ )) || true ;;
```
Comment (amp-rh) [inline ci-operator/step-registry/p2p/mtv-execute-live-migration/p2p-mtv-execute-live-migration-commands.sh]: > **Post-increment triggers errexit at zero**
>
> `(( failed++ ))` returns the value *before* incrementing. When `failed=0`, `((0))` evaluates to false (exit 1), requiring `|| true`. Pre-increment `((++failed))` returns 1 (true) and needs no guard.

```suggestion
               ((++failed)) ;;
```

File: ci-operator/step-registry/p2p/create-migration-test-vm/p2p-create-migration-test-vm-commands.sh [hunk 2]
```diff
@@ -11,28 +14,50 @@ set -euxo pipefail; shopt -s inherit_errexit
...
+# VmName — return the VM name for a 1-based index.
+# When vmCount=1 returns CNV_TEST_VM_NAME unchanged (backward compat).
+function VmName () {
+    typeset -i idx="${1:?}"; (($#)) && shift
+    if (( vmCount == 1 )); then
+        printf '%s' "${CNV_TEST_VM_NAME}"
+    else
+        printf 'test-vm-%d' "${idx}"
+    fi
+}
```
Comment (amp-rh) [inline ci-operator/step-registry/p2p/create-migration-test-vm/p2p-create-migration-test-vm-commands.sh:53]: > **Missing `true` at end of `VmName`**
>
> Best practice: every function must end with `true`. While `if`/`fi` normally returns 0, the explicit `true` guards against edge cases and satisfies the coding standard. Same applies to `VmName` in `p2p-mtv-execute-live-migration-commands.sh` (line 60).

```suggestion
    fi
    true
}
```

File: ci-operator/step-registry/p2p/mtv-execute-live-migration/p2p-mtv-execute-live-migration-commands.sh [hunk 3]
```diff
@@ -23,34 +33,49 @@ typeset -i sourceSpokeIndex="${MTV_SOURCE_SPOKE_INDEX}"
...
+# VmName — return the VM name for a 1-based index.
+# When vmCount=1 returns MTV_TEST_VM_NAME unchanged (backward compat).
+function VmName () {
+    typeset -i idx="${1:?}"; (($#)) && shift
+    if (( vmCount == 1 )); then
+        printf '%s' "${MTV_TEST_VM_NAME}"
+    else
+        printf 'test-vm-%d' "${idx}"
+    fi
+}
```
Comment (amp-rh) [inline ci-operator/step-registry/p2p/mtv-execute-live-migration/p2p-mtv-execute-live-migration-commands.sh:61]: > **Missing `true` at end of `VmName`**
>
> Same as the other script: every function must end with `true`.

```suggestion
    fi
    true
}
```

**Review by amp-rh — APPROVED** [review]
Solid multi-VM refactor. Three nits (all with inline suggestions), otherwise LGTM.

File: ci-operator/step-registry/p2p/create-migration-test-vm/p2p-create-migration-test-vm-commands.sh [hunk 4]
```diff
@@ -368,36 +412,51 @@ metadata:
...
-    case "${CNV_TEST_VM_IMAGE_TYPE}" in
-        cirros) ApplyCirrosDataVolume ;;
-        rhel)   ApplyRhelDataVolume ;;
-    esac
+        : "Creating VM ${i}/${vmCount}: ${currentVmName}"
 
-    WaitDataVolumeReady
+        if [[ "${CNV_TEST_VM_CLEAN}" == "true" ]] \
+            || SpokeOc get "datavolume/${currentDvName}" -n "${CNV_TEST_VM_NAMESPACE}" \
+                -o jsonpath='{.status.phase}' 2>/dev/null | grep -qE 'Failed|Lost'; then
```
Comment (sg-rh) [inline ci-operator/step-registry/p2p/create-migration-test-vm/p2p-create-migration-test-vm-commands.sh]: Best practices say never redirect `stderr `to `/dev/null`. The "not found" error is expected on first runs, but suppressing it hides unexpected errors too. Consider checking existence first or accepting the noise.

File: ci-operator/step-registry/p2p/create-migration-test-vm/p2p-create-migration-test-vm-commands.sh [hunk 5]
```diff
@@ -368,36 +412,51 @@ metadata:
...
+            printf '%s\n' "vm_count=${vmCount}"
             printf '%s\n' "vm_namespace=${CNV_TEST_VM_NAMESPACE}"
             printf '%s\n' "image_type=${CNV_TEST_VM_IMAGE_TYPE}"
             printf '%s\n' "spoke_kubeconfig=${spokeKubeconfig}"
-            SpokeOc get "virtualmachine/${CNV_TEST_VM_NAME}" "virtualmachineinstance/${CNV_TEST_VM_NAME}" \
-                "datavolume/${dvName}" "persistentvolumeclaim/${dvName}" \
-                -n "${CNV_TEST_VM_NAMESPACE}" -o wide
+            typeset -i j
+            for (( j = 1; j <= vmCount; j++ )); do
+                typeset vn dn
```
Comment (sg-rh) [inline ci-operator/step-registry/p2p/create-migration-test-vm/p2p-create-migration-test-vm-commands.sh]: Better to declare before the loop or use a function scope.

**Review by sg-rh — COMMENTED** [review]
Looks fine overall. below are some suggestions. 
I also noticed that the fn `VmName `is duplicated in both `p2p-create-migration-test-vm `and `p2p-mtv-execute-live-migration`. Pre-existing pattern for step-registry, but worth noting if naming logic changes.

**Review by sg-rh — APPROVED** [review]
/lgtm

**Review by sg-rh — APPROVED** [review]
/lgtm
